pyfar/classes/filter.py

```diff
@@ -434,33 +442,50 @@ def init_state(self, cshape, state='zeros'):
                 new_state[idx, ...] = spsignal.lfilter_zi(coeff[0], coeff[1])
         super().init_state(state=new_state)
 
+    def impulse_response_length(self):
+        """
+        Get the length of filter impulse response in samples.
+
+        The length is computed from the last non-zero coefficient per channel.
+
+        Returns
+        -------
+        impulse_response_length : array
+            An integer array of shape (C, ) containing the length in samples
+            where `C` denotes the number of channels of the filter.
+        """
+
+        # get filter coefficients
+        b = self.coefficients
+
+        # find last non-zero entry per channel
+        estimated_length = np.apply_along_axis(
+            lambda b: np.max(np.nonzero(b)), axis=-1, arr=b) + 1
+
+        # restore input data shape
+        estimated_length = np.reshape(estimated_length, b.shape[:-1])
+
+        return estimated_length.astype(int)
+
     def impulse_response(self, n_samples=None):
```

Comment:
do we need to overwirte it? its the same as Filter.impulse_response() and impulse_response_length is implemented for each class.

Reply:
Yes we need to and decided to do so in the weekly. The reason is that the calculation of the impulse response length is different for each filter, hence we need individual docs for each filter class.